src/k8s/pkg/k8sd/setup/k8s_apiserver_proxy_test.go

```diff
@@ -130,4 +130,29 @@ func TestK8sApiServerProxy(t *testing.T) {
 		// Compare the expected endpoints with those in the file
 		g.Expect(config.Endpoints).To(Equal(endpoints))
 	})
+
+	t.Run("IPv6", func(t *testing.T) {
+		g := NewWithT(t)
+
+		// Create a mock snap
+		s := mustSetupSnapAndDirectories(t, setKubeletMock)
+		s.Mock.Hostname = "dev"
+
+		g.Expect(setup.K8sAPIServerProxy(s, nil, "[2001:db8::]", nil)).To(Succeed())
+
+		tests := []struct {
+			key         string
+			expectedVal string
+		}{
+			{key: "--listen", expectedVal: "[2001:db8::]:6443"},
+		}
+		for _, tc := range tests {
+			t.Run(tc.key, func(t *testing.T) {
```

Comment:
nit: Maybe we can have a `tc.name` here instead. If we decided to go with this, we might need to change the rest of the tests as well.

Reply:
not exactly sure what you mean here. Do you want to rename the `tc.key` to `tc.name` or do you want to add a `tc.name` field? If the latter, what value do you see in that?

Reply:
Hmm you're right, there's no real value in that.